### PR TITLE
Minor cleanups.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

- in backend_ps: when writing to a unicode file, no need to first read the entire tempfile in memory and then decode it; just do everything in one step.
- in type1font: str.split() already strips spaces at the edges of the string (see https://docs.python.org/3/library/stdtypes.html#str.split).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_ps.py [hunk 1]
```diff
@@ -1354,12 +1352,10 @@ def write(self, *kl, **kwargs):
...
             if is_writable_file_like(outfile):
-                if file_requires_unicode(outfile):
-                    with io.open(tmpfile, 'rb') as fh:
-                        outfile.write(fh.read().decode('latin-1'))
-                else:
-                    with io.open(tmpfile, 'rb') as fh:
-                        outfile.write(fh.read())
+                with (io.open(tmpfile, 'r', encoding='latin-1')
+                        if file_requires_unicode(outfile)
+                        else io.open(tmpfile, 'rb')) as fh:
```
Comment (efiring) [inline lib/matplotlib/backends/backend_ps.py]: At the risk of bikeshedding... I find this 3-line construct a bit of a trip hazard when reading.  How about this:
```python
if file_requires_unicode(outfile):
    kwd = dict(mode='r', encoding='latin-1')
else:
    kwd = dict(mode='rb')
with io.open(tmpfile, **kwd) as fh:
```
Or if you want to reduce the line count:
```python
kwd = (dict(mode='r', encoding='latin-1') if file_requires_unicode(outfile)
            else dict(mode='rb'))
with io.open(tmpfile, **kwd) as fh:
```
The latter option is no longer than your original, and to me, much more readable. I haven't checked, but I hope it doesn't fall afoul of the line length limit.

Comment (jklymak) [reply lib/matplotlib/backends/backend_ps.py]: Yeah, I totally agree w/ this but thought maybe I was being an anti-fancy-python-tricks stick-in-the-mud

Comment (efiring) [reply lib/matplotlib/backends/backend_ps.py]: OK

Comment (QuLogic) [reply lib/matplotlib/backends/backend_ps.py]: Was this actually changed? It's not marked outdated.

Comment (anntzer) [reply lib/matplotlib/backends/backend_ps.py]: Ah, sorry.  There were two instances of it, only changed one of them.

**Pull request merged**